### ISSUE TITLE
Improper Use of Link rule and anchors without href (discussion needed)

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#1694]
Driscoll has this code flagging an improper use of link:
`<a id="meet-the-team"></a>`

Technically, this is valid HTML. [W3 says:](https://www.w3.org/TR/html401/struct/links.html#h-12.2.3)
>The following example names a destination anchor with the [id](https://www.w3.org/TR/html401/struct/global.html#adef-id) attribute:

> I just returned from vacation! Here's a
>`<A id="anchor-two">photo of my family at the lake.</A>.`

I don't like using an anchor as the landing point on a jump link when you could use something else with an `id`, since it would likely still be styled visually like a link, though it would do nothing when clicked and would not be keyboard focusable. 

This possible user confusion may be reason enough to leave our rule alone, however since this is technically valid HTML, I wanted to open discussion on whether our rule should be modified.

**Comment by SteveJonesDev** [comment]
I think we can add a narrow exception here. An empty `<a id="foo"></a>` used purely as a jump target is fine, no `href` means it's not keyboard focusable, and per the HTML spec its implicit ARIA role is `generic` not `link`, so assistive technologies won't announce it as one. No real accessibility concern. But if it has text content `<a id="foo">Meet the Team</a>`, it should still fail since it looks like a link but does nothing.

Fix in this PR, https://github.com/equalizedigital/accessibility-checker/pull/1698

**Comment by amberhinds** [comment]
That sounds like a good approach. We should check to make sure that `<a id="foo"></a>` doesn't also flag an empty link.
